applications/incompressible_navier_stokes/periodic_hill/application.h

```diff
@@ -251,8 +276,11 @@ class Application : public ApplicationBase<dim, Number>
       this->param.order_time_integrator <= 2 ? this->param.order_time_integrator : 2;
 
     this->param.solver_momentum         = SolverMomentum::CG;
+
     this->param.solver_data_momentum    = SolverData(1000, 1.e-12, 1.e-3);
-    this->param.preconditioner_momentum = MomentumPreconditioner::PointJacobi;
+    this->param.preconditioner_momentum = spatial_discretization == SpatialDiscretization::L2 ?
```

Comment:
is the `MomentumPreconditioner::InverseMassMatrix` also asserted in this branch? @kronbichler you think a global solver is in general not competitive?

Reply:
Regarding the assertion, I need to check specifically. Regarding the global solver: If there is no cheap and accurate mass matrix inversion like for H(div), it makes more sense to use the diagonal of the entire operator and not just try to approximate the mass operator. For example, if one preconditions by the inverse mass matrix, but that itself is an iterative solver, I see little point in nesting these solvers, and it should make more sense to directly apply the diagonal of the momentum operator as a preconditioner in a single GMRES solver.